In [5]:
import pandas as pd

In [6]:
df = pd.read_csv("ecommerce_sales_customer_behavior.csv")
df

,order_id,year,month,quarter,is_weekend_order,platform,country,customer_segment,customer_age_group,customer_gender,...,return_reason,customer_satisfaction,session_duration_minutes,pages_viewed,cart_abandoned_before,wishlist_used,coupon_used,subscription_member,repeat_purchase_count,loyalty_points
0,ORD0000001,2022,4,Q2,Yes,AliExpress,Germany,New Customer,35-44,Male,...,NaN,Dissatisfied,78.7,2,Yes,No,No,No,0,0
1,ORD0000002,2023,6,Q2,Yes,Flipkart,Australia,Loyal Customer,35-44,Male,...,NaN,Neutral,18.2,7,No,Yes,No,No,9,198
2,ORD0000003,2022,2,Q1,Yes,Amazon,Turkey,VIP Customer,45-54,Male,...,NaN,Very Dissatisfied,5.1,1,No,No,No,Yes,30,1020
3,ORD0000004,2022,7,Q3,No,AliExpress,USA,New Customer,35-44,Male,...,NaN,Satisfied,22.0,6,Yes,No,No,Yes,0,0
4,ORD0000005,2024,4,Q2,No,Daraz,Australia,Loyal Customer,35-44,Female,...,NaN,Satisfied,4.5,8,Yes,No,Yes,No,13,598
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
9995,ORD0009996,2022,3,Q1,No,Amazon,Australia,VIP Customer,45-54,Male,...,NaN,Very Satisfied,14.1,6,No,No,No,Yes,47,2021
9996,ORD0009997,2021,11,Q4,No,Amazon,India,VIP Customer,45-54,Female,...,NaN,Satisfied,44.5,8,No,No,No,No,37,814
9997,ORD0009998,2021,10,Q4,Yes,eBay,Pakistan,Loyal Customer,65+,Male,...,NaN,Satisfied,8.9,8,Yes,No,No,No,5,50
9998,ORD0009999,2023,2,Q1,Yes,eBay,Bangladesh,New Customer,35-44,Female,...,NaN,Satisfied,62.4,4,No,Yes,Yes,No,0,0


El dataset tiene datos de pedidos de una tienda online donde incluye cuándo se hizo el pedido(fecha), quién es el cliente (país, edad, género, tipo de cliente), qué categoria de producto compro y cuánto pagó (con descuentos, impuestos y envío) y si quedó satisfecho o lo devolvió. También hay información de cómo navegó en la página antes de comprar y si es un cliente que compra seguido.

In [26]:
df[df["final_amount_usd"] > 500].shape[0]

3209

1. 3209 pedidos tienen un monto final superior a 500USD

In [27]:
n = df["customer_segment"].value_counts(1)
n

customer_segment
Returning Customer    0.2977
New Customer          0.2522
Loyal Customer        0.2166
VIP Customer          0.1213
Inactive Customer     0.1122
Name: proportion, dtype: float64

2. El 12% de los pedidos realizados son por VIP Customer

In [28]:
df.query("customer_age_group == '25-34' and customer_gender == 'Female'").shape[0]



1471

3. 1471 pedidos fueron hechos por mujeres entre 25-34 años.

In [29]:
cf = df.query("(customer_satisfaction == 'Dissatisfied' or customer_satisfaction == 'Very Dissatisfied') and returned == 'Yes'")
cf["product_category"].value_counts()



product_category
Electronics               12
Fashion & Apparel          9
Beauty & Personal Care     8
Sports & Fitness           8
Mobile & Tablets           7
Home & Kitchen             6
Toys & Games               5
Health & Wellness          4
Books & Education          4
Jewelry & Accessories      2
Automotive                 2
Food & Grocery             2
Name: count, dtype: int64

In [31]:
cf["product_category"].shape[0]

69

4. Fueron devueltos 69 pediodos con calificación "Dissatisfied" ó "Very Dissatisfied" siendo Electronics el tipo de producto que mas fue devuelto con 12.

In [32]:
f= df.query("is_weekend_order == 'Yes'")
f[f["coupon_used"] == "Yes"].shape[0]/f.shape[0]

0.35851035929714137

5.El 35.8% de los pedidos del fin de semana utilizo un cupon de descuento.

In [34]:
j=df.query("customer_segment == 'Inactive Customer' and discount_pct > 30")
j["total_paid_usd"].mean()

np.float64(403.40180555555554)

In [36]:
j.shape[0]

72

6. 72 pedidos cumplen con "Iniciative Customer" y un descuento mayor a 30% siendo 403USD el monto promedio pagado.

In [52]:
df.columns

Index(['order_id', 'year', 'month', 'quarter', 'is_weekend_order', 'platform',
       'country', 'customer_segment', 'customer_age_group', 'customer_gender',
       'acquisition_channel', 'device_used', 'product_category',
       'unit_price_usd', 'quantity_ordered', 'order_value_usd', 'discount_pct',
       'discount_amount_usd', 'final_amount_usd', 'shipping_cost_usd',
       'tax_amount_usd', 'total_paid_usd', 'payment_method', 'shipping_method',
       'delivery_days', 'product_rating', 'review_submitted', 'returned',
       'return_reason', 'customer_satisfaction', 'session_duration_minutes',
       'pages_viewed', 'cart_abandoned_before', 'wishlist_used', 'coupon_used',
       'subscription_member', 'repeat_purchase_count', 'loyalty_points'],
      dtype='str')

## 7 Resumen ejecutivo

In [39]:
columnas_resumen_ejecutivo = [
    "country",
    "customer_segment",
    "platform",
    "product_category",
    "year",
    "month",
    "quantity_ordered",
    "total_paid_usd",
]

df_ejecutivo = df[columnas_resumen_ejecutivo]
df_ejecutivo



,country,customer_segment,platform,product_category,year,month,quantity_ordered,total_paid_usd
0,Germany,New Customer,AliExpress,Home & Kitchen,2022,4,2,456.39
1,Australia,Loyal Customer,Flipkart,Beauty & Personal Care,2023,6,1,149.02
2,Turkey,VIP Customer,Amazon,Electronics,2022,2,3,5783.94
3,USA,New Customer,AliExpress,Health & Wellness,2022,7,1,221.94
4,Australia,Loyal Customer,Daraz,Automotive,2024,4,1,344.71
...,...,...,...,...,...,...,...,...
9995,Australia,VIP Customer,Amazon,Mobile & Tablets,2022,3,1,965.01
9996,India,VIP Customer,Amazon,Food & Grocery,2021,11,5,292.83
9997,Pakistan,Loyal Customer,eBay,Books & Education,2021,10,1,61.78
9998,Bangladesh,New Customer,eBay,Food & Grocery,2023,2,2,89.92


In [43]:
ds = df.assign(ahorro_pct_real = df["discount_amount_usd"]* 100 / df["order_value_usd"])
ds[["discount_pct","ahorro_pct_real"]]

,discount_pct,ahorro_pct_real
0,20,20.000371
1,25,25.000000
2,15,15.000000
3,5,4.998836
4,30,29.999750
...,...,...
9995,10,10.000309
9996,30,29.998613
9997,25,25.007634
9998,10,9.995124


In [46]:
ds["comparacion"] = ds["discount_pct"]-ds["ahorro_pct_real"]
ds["comparacion"].describe()

count    10000.000000
mean         0.000038
std          0.002472
min         -0.036049
25%         -0.000153
50%          0.000000
75%          0.000163
max          0.036203
Name: comparacion, dtype: float64

8.Al ver la comparación llegamos a la conclusón de ambas columnas tienen el mismo resultado con unas milesimas de diferencia.Esto indica que las dos significan lo mismo y que ambas fueron calculadas con la misma logica.

In [44]:
fp = df.assign(compra_grande = df["quantity_ordered"] > 2)
cp = fp.query("compra_grande == True").shape[0]
c = fp.query("compra_grande == True and returned == 'Yes'").shape[0]
c/cp


0.06420133538777606

9. Un 6.42% de las compras grandes resulto en devolución.

In [47]:
top = df.sort_values("total_paid_usd", ascending = False).head(10)
top[["country", "platform", "product_category", "total_paid_usd"]]


,country,platform,product_category,total_paid_usd
6933,Nigeria,Amazon,Electronics,11829.02
9120,UK,Flipkart,Electronics,10672.50
1904,Turkey,Daraz,Electronics,10566.55
3537,Pakistan,Jumia,Electronics,9558.47
5843,UAE,Amazon,Electronics,8996.51
4292,Pakistan,Jumia,Electronics,8975.51
960,Egypt,Flipkart,Electronics,8817.56
153,UAE,AliExpress,Electronics,8570.46
8044,Canada,Etsy,Electronics,8542.20
13,UAE,AliExpress,Electronics,8141.50


10. Al ordenar el dataset de forma descendente nos damos cuenta que todos los productos pertenecen a la categoria de "Electronics". Lo que nos puede indicar que este tipo de producto es el mas caro dentro del catalogo.

In [49]:
pf = (df.groupby("platform")
      .agg( promedio_pagado_plataforma = ("total_paid_usd","mean"))
      .sort_values("promedio_pagado_plataforma", ascending=False))
pf

,promedio_pagado_plataforma
platform,
Daraz,684.270947
AliExpress,679.233179
Amazon,676.490409
Flipkart,666.103077
WooCommerce,664.837305
Jumia,662.281364
Shopify Store,645.465098
Etsy,632.136624
eBay,613.911184


11. La plataforma Daraz es la que genera el mayor ingreso promedio por pedido. Sin embargo, la diferencia con las demás plataformas no es muy grande, por lo que la plataforma no parece ser el factor más determinante en el valor del pedido.

12. Las preguntas más importantes son la 9 y la 11. La pregunta 9 es importante porque conocer el porcentaje de compras grandes que terminan en devolución permite identificar si existe un patrón de riesgo asociado a pedidos de varias unidades, por ejemplo,con errores al empacar. Y la 11, porque aunque la diferencia entre plataformas no es muy grande, saber cuál deja en promedio más ingreso por pedido puede ayudar a la empresa a decidir en qué plataforma conviene invertir más o negociar mejores condiciones.